vllm/attention/backends/flash_attn.py

```diff
@@ -259,7 +259,11 @@ def _add_seq_group(
                 block_table = block_tables[seq_id]
             elif ((chunked_prefill_enabled or not is_prompt)
                   and block_tables is not None):
-                block_table = block_tables[seq_id][-curr_sliding_window_block:]
+                if curr_sliding_window_block == 0:
+                    block_table = block_tables[seq_id]
+                else:
+                    block_table = block_tables[seq_id][
+                        -curr_sliding_window_block:]
```

Comment:
Out of curiosity, why slicing is slow?

Reply:
iirc it always creates a copy

Reply:
yeah, slicing is a bit confusing, since in numpy it is actually a view (and not a copy).

Reply:
small simplification nit, could change to:
```python
                block_table = block_tables[seq_id]
                if curr_sliding_window_block != 0:
                    block_table = block_table[-curr_sliding_window_block:]
```